lib/src/configuration.cc

```diff
@@ -30,41 +30,40 @@ namespace lth_log = leatherman::logging;
 
 #ifdef _WIN32
     namespace lth_w = leatherman::windows;
-    const std::string DEFAULT_ACTION_RESULTS_DIR = []() {
-        wchar_t buf[MAX_PATH+1];
-        auto num = GetTempPathW(MAX_PATH+1, buf);
-        if (num <= 0 || num > MAX_PATH) {
-            throw std::runtime_error(
-                (boost::format("failure getting Windows TEMP directory: %1%")
-                    % lth_w::system_error()).str());
-        }
-        fs::path p = fs::path(buf) / "pxp-agent";
-        return p.string() + "/";
-    }();
-
-    static const fs::path DEFAULT_SHARE_DIR { "TBD" };
-
-    static const fs::path DEFAULT_CONF_DIR = []() {
+    static const fs::path DATA_DIR = []() {
         wchar_t szPath[MAX_PATH+1];
         if (FAILED(SHGetFolderPathW(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
-            throw std::runtime_error(
+            throw Configuration::Error {
                 (boost::format("failure getting Windows AppData directory: %1%")
-                    % lth_w::system_error()).str());
+                    % lth_w::system_error()).str() };
         }
         return fs::path(szPath) / "PuppetLabs" / "pxp-agent";
     }();
-#else
-    const std::string DEFAULT_ACTION_RESULTS_DIR = "/tmp/pxp-agent/";
 
-    static const fs::path DEFAULT_SHARE_DIR { "/opt/puppetlabs/pxp-agent" };
+    static const fs::path DEFAULT_CONF_DIR { DATA_DIR / "etc" };
+    const std::string DEFAULT_SPOOL_DIR { (DATA_DIR / "var" / "spool").string() };
+    static const std::string PID_DIR { (DATA_DIR / "var" / "run").string() };
+    static const std::string DEFAULT_LOG_DIR { (DATA_DIR / "var" / "log").string() };
+#else
     static const fs::path DEFAULT_CONF_DIR { "/etc/puppetlabs/pxp-agent" };
+    const std::string DEFAULT_SPOOL_DIR { "/opt/puppetlabs/pxp-agent/spool" };
+    static const std::string PID_DIR { "/var/run/puppetlabs" };
+    static const std::string DEFAULT_LOG_DIR { "/var/log/puppetlabs/pxp-agent" };
 #endif
 
-static const std::string DEFAULT_MODULES_DIR = (DEFAULT_SHARE_DIR / "modules").string();
-static const std::string DEFAULT_MODULES_CONF_DIR = (DEFAULT_CONF_DIR / "modules.d").string();
-static const std::string DEFAULT_CONFIG_FILE = (DEFAULT_CONF_DIR / "pxp-agent.cfg").string();
+
+// TODO(ale): temporary fix; update this for CTH-373
+static const std::string DEFAULT_MODULES_DIR {
+    (fs::current_path() / "modules").string() };
```

Comment:
We've done that through CMAKE defines in packaging for Facter (https://github.com/puppetlabs/puppet-agent/blob/master/configs/components/facter.rb#L157).